airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -153,41 +153,42 @@ public SourceSpecificationRead getSourceSpecification(SourceIdRequestBody source
     UUID sourceId = sourceIdRequestBody.getSourceId();
     StandardSource source = configRepository.getStandardSource(sourceId);
     final String imageName = DockerUtils.getTaggedImageName(source.getDockerRepository(), source.getDockerImageTag());
-    final long jobId = schedulerPersistence.createGetSpecJob(imageName);
-    LOGGER.debug("getSourceSpec jobId = {}", jobId);
-
-    Job job = waitUntilJobIsTerminalOrTimeout(jobId);
+    final ConnectorSpecification spec = getConnectorSpecification(imageName);
 
     TrackingClientSingleton.get().track("get_source_spec", ImmutableMap.<String, Object>builder()
-        .put("source_id", sourceId)
+        .put("source_id", source.getSourceId())
+        .put("source_name", source.getName())
         .put("image_name", imageName)
-        .put("job_id", jobId)
         .build());
 
-    final ConnectorSpecification spec = job.getOutput().orElseThrow().getGetSpec().getSpecification();
     return new SourceSpecificationRead()
         .connectionSpecification(spec.getConnectionSpecification())
         .documentationUrl(spec.getDocumentationUrl().toString())
         .sourceId(sourceId);
   }
 
+  ConnectorSpecification getConnectorSpecification(String imageName) throws IOException {
```

Comment:
why is this visible?

Reply:
we need a way to get the spec using only an imageName. The other methods (getSourceSpec/destinationSpec) require a source ID or destination ID. This is the method I'm exposing to allow ID-less retrieval of specs. Do you see an issue with this pattern?